File: src/main/java/com/example/demo/user/account/UserRepository.java
package com.example.demo.user.account;

import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface UserRepository extends JpaRepository<UserAccount, String> {

    Optional<UserAccount> findByUsername(String username);

    Optional<UserAccount> findByUsernameAndDeletedAtIsNull(String username);

    Optional<UserAccount> findByIdAndDeletedAtIsNull(String id);

    Optional<UserAccount> findByActiveNicknameKey(String activeNicknameKey);

    boolean existsByActiveUsernameKey(String activeUsernameKey);

    boolean existsByActiveNicknameKey(String activeNicknameKey);
}

File: src/main/java/com/example/demo/user/account/UserRepository.kt
package com.example.demo.user.account

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.Optional

@Repository
interface UserRepository : JpaRepository<UserAccount, String> {
    fun findByUsername(username: String?): Optional<UserAccount>

    fun findByUsernameAndDeletedAtIsNull(username: String?): Optional<UserAccount>

    fun findByIdAndDeletedAtIsNull(id: String?): Optional<UserAccount>

    fun findByActiveNicknameKey(activeNicknameKey: String?): Optional<UserAccount>

    fun existsByActiveUsernameKey(activeUsernameKey: String?): Boolean

    fun existsByActiveNicknameKey(activeNicknameKey: String?): Boolean
}
